### PR TITLE
php8 support on Suse 15.4

### PR DESCRIPTION
**Pull request by CaptainSifff**
Hi all,
this enables php8 support on Suse 15.4.
I introduced a switch that allows you to use php7 on 15.4 (I believe). 
I tested it in my environment for configuring php8 and it seems to work so far.
Please tell me what you think about it.

### CODE REVIEW

**Comment by grisu48** [comment]
Hey! In general a very good idea, but perhaps it's better to have a php7 and a separate php8 role :thinking:

**Comment by CaptainSifff** [comment]
But wouldn't this mean, that we would also need to have this php7 and php8 distinction on all distributions?

**Comment by grisu48** [comment]
> But wouldn't this mean, that we would also need to have this php7 and php8 distinction on all distributions?

If php7 and php8 are available, then probably yes, unless you have a better idea. When I wrote this only php7 was there, so there was no need to separate it. Last time I checked, there was php7 and php8 available to be installed, so we probably have to split the role into two ones.

Or what do you think?

**Comment by CaptainSifff** [comment]
this means we have a php7 role compatible with opensuse 15.3 and 15.4 and a php8 role only for 15.4 . This makes for a clear separation, but has quite some duplicate code.

**Comment by grisu48** [comment]
I'm afraid so, although the duplication would only endure for the lifetime of 15.4.

Perhaps we update this role to support always the most recent version (here php8) and create a separate "legacy" role if needed.

**Comment by CaptainSifff** [comment]
I think the easiest maintainable thing would be to define that the role by default installs the most recent php version on an opensuse distribution. We can keep the configuration switch, then people can tinker....

**Comment by grisu48** [comment]
After re-thinking about it, I agree.

**Comment by CaptainSifff** [comment]
OK, once I have tested that I can utilize this with my drupal9 install and my nextcloud setup, I will prepare it again for merging. For now there's still the change in my branch to make the pool name configurable. (Different pools for drupal and for nextcloud)
Would there be interest in a letsencrypt role(finished), a drupal9 role(final test) and a nextcloud role(planned)?

**Comment by grisu48** [comment]
> Would there be interest in a letsencrypt role(finished), a drupal9 role(final test) and a nextcloud role(planned)?

YES!!! Absolutely! :partying_face: Please do!

**Comment by CaptainSifff** [comment]
So, the letsencrypt role is now public: https://github.com/GeekOops/geekoops-letsencrypt-apache

File: tasks/main.yml [hunk 1]
```diff
@@ -35,14 +35,24 @@
...
     mode: 0644
   notify: restart php-fpm
   tags: ['php-fpm']
   when: configure_php_fpm == true
 
+- name: Ensure AppArmor knows about our new config file and the socket it tries to create
+  ansible.builtin.blockinfile:
+    path: /etc/apparmor.d/local/php-fpm
+    state: present
+    block: |
```
Comment (grisu48) [inline tasks/main.yml]: Trailing space

**Review by grisu48 — COMMENTED** [review]
One small linting error. `molecule` also complains about too long lines, so could you please change the following in `molecule/default/molecule.yml`?

```
lint: |
  yamllint -d "{extends: relaxed, rules: {line-length: disable}}" .
```

This disables the line-length check for the yaml files.

In addition, the Leap 15.4 molecule test runs are currently failing due to some container issues, which I'm investigating right now.

Could you please also rebase your commits into one single one? That would be great, then we can proceed with this awesome contribution!

**Comment by grisu48** [comment]
Please rebase all commits to one, then we can merge:

```
git rebase -i master
```

And then select `fixup` on all commits but the first one.

**Comment by CaptainSifff** [comment]
like this?

**Comment by grisu48** [comment]
> like this?

This is awesome. Thank you! :rocket:

**Pull request merged**